### PR TITLE
🦺 zx: Allow unrestricted property names in zbus-xmlgen

### LINKED ISSUE #550: xmlgen: special characters in property names
zbus-xmlgen currently enforces member name restrictions for d-bus properties. It does not allow '-' in property names.

The D-Bus specification (https://dbus.freedesktop.org/doc/dbus-specification.html#standard-interfaces-properties) states

> Strictly speaking, D-Bus property names are not required to follow the same naming restrictions as member names

It also states properties that don't follow those restrictions should be avoided but that's a different topic.
Also, they are actually used in the wild, e.g. by gnome-session (https://gitlab.gnome.org/GNOME/gnome-session/-/blob/main/gnome-session/org.gnome.SessionManager.Presence.xml?ref_type=heads#L36).

To follow the specification, I think zbus should allow them.

### PR DESCRIPTION
**Pull request by Cobinja**
This fixes #550 by adding a struct PropertyName to zbus-names. It's essentially a copy of the already existing struct MemberName but only checks for the name length instead of enforcing all name restrictions.
Also added/changed the appropriate tests.

### CODE REVIEW

File: zbus_names/src/property_name.rs
Comment (zeenix) [inline zbus_names/src/property_name.rs:15]: * nit: why `in`? I typically abbreviate the phrase, so `pn` would suffice.
* `(method or signal)` doesn't make sense here. I guess you based this on the copy of `MemberName` and just forgot to update the docs? :wink: 
* `an` -> `a`.

**Review by zeenix — CHANGES_REQUESTED** [review]
Thanks so much for the work @Cobinja !  Very nice. Apart from a few minor issues, kindly follow [our guidelines](https://github.com/dbus2/zbus/blob/main/CONTRIBUTING.md#submitting-pull-requests) regarding commit hygiene?

File: zbus_names/src/property_name.rs
Comment (Cobinja) [reply zbus_names/src/property_name.rs]: oh, yeah, just forgot a doc-update. Done in the new commit.

Comment (zeenix) [reply zbus_names/src/property_name.rs]: > oh, yeah, just forgot a doc-update.

no problem.

> Done in the new commit.

Actually, it's a fixup to an unmerged commit so it should be squashed into the previous commit as per the contribution guideline. The main commit contains 2 logical changes though (1. addition of new API in zbus_names and 2. the use of it in the zbus_xmlgen) so it should be split into 2.

Oh and please rebase on the main branch, not merge it in so the PR only shows commits of this PR. You'll want to rebase anyway for the squashing.

Good luck and don't feel back if any of this is new to you and get stuck. I can always help with any/all of this. Just let me know if you need help.

Comment (Cobinja) [reply zbus_names/src/property_name.rs]: Should tests go in a separate commit as well? Also, was the push for tokio-vsock-test to my branch accidently? I assume it should have gone to your main branch instead.

Comment (zeenix) [reply zbus_names/src/property_name.rs]: > Should tests go in a separate commit as well?

Would be even better IMO but I leave that to you.

>  Also, was the push for tokio-vsock-test to my branch accidently? I assume it should have gone to your main branch instead.

This happens when you merge the main branch into your (PR) branch. You may have done that indirectly by selecting the "update the branch with main" option Github offers (I wonder if there is a way to disable that) or manually, I can't tell. 

In any case, the solution is simple: `git remote update origin && git rebase origin/main YOUR_BRANCH `. Then `git reset origin/main` to redo the commits. Use `git add -p` to stage the changes for the first commit and `git commit` to create the commit for staged changes only. Then repeat for the second commit. Once done with all that, `git push -f YOUR_REMOTE YOUR_BRANCH` to finally push your commits.

Comment (Cobinja) [reply zbus_names/src/property_name.rs]: > 
> This happens when you merge the main branch into your (PR) branch. You may have done that indirectly by selecting the "update the branch with main" option Github offers (I wonder if there is a way to disable that) or manually, I can't tell.
> 
In that case, github did something it shouldn't, because I didn't do anything to my branch via github. I branched from the main branch today and pushed without fetching, merging or pulling in between. And the push was not with github-desktop or github-cli or so, but plain git-cli.

Comment (zeenix) [reply zbus_names/src/property_name.rs]: Very strange. I just checked and the commit in question was merged into git main 2 weeks ago. So I don't know how it ended up in your branch.

Anyway, a rebase should solve the issue.

Comment (Cobinja) [reply zbus_names/src/property_name.rs]: New force-push with the same content as before, separated into 3 commits (1 for zbus-names, one for zbus-xml and one for the tests). It seems to have solved the issue with the stray commit.

Comment (zeenix) [reply zbus_names/src/property_name.rs]: > New force-push with the same content as before, separated into 3 commits (1 for zbus-names, one for zbus-xml and one for the tests).

Awesome! No emojis? :sob: I can add them for you, if you like?

> It seems to have solved the issue with the stray commit.

I thought it would. :)

Comment (zeenix) [reply zbus_names/src/property_name.rs]: > Add them if you want to. ¯_(ツ)_/¯

Sure but please do add them in the future contributions. :) I also see now that you didn't quite follow the commit message guidelines/conventions. I'll fix that as well. Please don't make any changes until I'm done..

Comment (zeenix) [reply zbus_names/src/property_name.rs]: Done! Please have a look at the commits for future reference.

Comment (zeenix) [reply zbus_names/src/property_name.rs]: Oh, the formatting is all wrong. I'll fix that too.

Comment (zeenix) [reply zbus_names/src/property_name.rs]: fixed! I'm done. Please resolve this thread once you've read my comments here.

**Review by zeenix — APPROVED** [review]

**Pull request merged**
